dds/DCPS/ReactorInterceptor.h

```diff
@@ -68,6 +69,8 @@ class OpenDDS_Dcps_Export ReactorInterceptor : public RcEventHandler {
   virtual ~ReactorInterceptor();
   int handle_exception(ACE_HANDLE /*fd*/);
   int handle_exception_i(ACE_Guard<ACE_Thread_Mutex>& guard);
+  void process_command_queue_i(ACE_Guard<ACE_Thread_Mutex>& guard);
```

Comment:
Passing around `ACE_Guard` doesn't seem to be necessary.  The `process_command_queue` function (whatever its name ends up being) should handle it internally.
Pseudocode:
```
while (true) {
  CommandPtr p;
  {
    LOCK
    if (empty) return;
    p = front; pop;
  }
  if (p) run_it;
}
```

Reply:
I was trying to follow the convention of handle_exception_i, which is called by more than just handle_exception and presumably uses the guard to signify that the method should have the lock locked when it's called (I don't think it was using the guard either). I think it would also get optimized out by most compilers... but if you want to change it I'm OK with that.

Reply:
Also, process_command_queue seems to always assume that the lock will be locked at the beginning and end of the method (see wait(), which calls it via handle_exception_i), so the reverse lock makes sense. I agree there's no real purpose to passing the guard (unless perhaps we made both public and private versions, and even then it's just a reminder to developers they should have a guard in place before they call the internal version).